### PR TITLE
In-app recommendation: XML for pom.xml

### PR DESCRIPTION
**Pull request by akaroml**

### CODE REVIEW

File: src/recommendation/xml.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import { validateAndRecommendExtension } from "../utils";
+import { instrumentCommand } from "../command";
+
+const KEY_XML_RECOMMENDATION_LAST_SHOW_TIME = "xmlRecommendationLastShowTime";
+
+export function initialize (context: vscode.ExtensionContext) {
+  const recommendationHandler = instrumentCommand(context, "recommendExtension", extensionRecommendationHandler);
+  context.subscriptions.push(vscode.workspace.onDidOpenTextDocument(e => {
+    if (e.uri.path.toLowerCase().indexOf("pom.xml") !== -1) {
```
Comment (testforstephen) [inline src/recommendation/xml.ts]: => `if (e.fileName === "pom.xml")`

File: src/utils/index.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import * as vscode from "vscode";
+
+export async function validateAndRecommendExtension(extName: string, message: string): Promise<boolean> {
+  const ext = vscode.extensions.getExtension(extName);
+  if (ext) {
+    return true;
+  }
+
+  const action = "Details";
```
Comment (testforstephen) [inline src/utils/index.ts]: Should add an `Install` button in the recommend dialog.

File: src/recommendation/xml.ts (hunk 1 above)
Comment (Eskibear) [reply src/recommendation/xml.ts]: `e.filename.toLowerCase()` for robustness? I've seen `POM.XML` in some projects.

File: src/utils/index.ts (hunk 2 above)
Comment (akaroml) [reply src/utils/index.ts]: There is no API in vscode which allows you to "Install" an extension directly. So the workaround is to show the "Details" page and then you can click the "Install" button to achieve that.

File: src/recommendation/xml.ts (hunk 1 above)
Comment (akaroml) [reply src/recommendation/xml.ts]: e.fileName is the full path. === sign cannot be used directly to check whether the current file is pom.xml

File: src/utils/index.ts (hunk 2 above)
Comment (testforstephen) [reply src/utils/index.ts]: VS Code's recommendation has the `Install` button, could we ask for their implementation?

**Pull request merged**